packages/zoe/src/zoeService/zoe.js

```diff
@@ -0,0 +1,357 @@
+        getPublicFacet: () => publicFacetPromiseKit.promise,
+        getTerms: () => instanceRecord.terms,
+        getIssuers: () => instanceRecord.issuerKeywordRecord,
+        getBrands: () => instanceRecord.brandKeywordRecord,
+        getInstance: () => instance,
+      };
+
+      instanceToInstanceAdmin.init(instance, instanceAdmin);
+
+      /** @type {ZoeInstanceAdmin} */
+      const zoeInstanceAdminForZcf = {
+        makeInvitation: (invitationHandle, description, customProperties) => {
+          const invitationAmount = invitationAmountMath.make(
+            harden([
+              {
+                ...customProperties,
+                description,
+                handle: invitationHandle,
+                instance,
+                installation,
+              },
+            ]),
+          );
+          return invitationMint.mintPayment(invitationAmount);
+        },
+        // checks of keyword done on zcf side
+        saveIssuer: (issuerP, keyword) =>
+          issuerTable
+            .getPromiseForIssuerRecord(issuerP)
+            .then(({ issuer, brand }) => {
+              registerIssuerByKeyword(keyword, issuer, brand);
+              if (!brandToPurse.has(brand)) {
+                brandToPurse.init(brand, E(issuer).makeEmptyPurse());
+              }
+            }),
+        // A Seat requested by the contract without an offer
+        makeOfferlessSeat: (initialAllocation, proposal) => {
+          const { userSeat, notifier, zoeSeatAdmin } = makeZoeSeatAdminKit(
+            initialAllocation,
+            instanceAdmin,
+            cleanProposal(getAmountMath, proposal),
+            brandToPurse,
+          );
+          zoeSeatAdmins.add(zoeSeatAdmin);
+          return { userSeat, notifier, zoeSeatAdmin };
+        },
+        shutdown: () => {
+          exitAllSeats();
+          adminNode.terminate();
+        },
+        makeZoeMint,
+      };
+
+      // At this point, the contract will start executing. All must be ready
+
+      const {
+        creatorFacet = {},
+        publicFacet = {},
+        creatorInvitation,
+        addSeatObj,
+      } = await E(zcfRoot).executeContract(
+        bundle,
+        zoeService,
+        invitationIssuer,
+        zoeInstanceAdminForZcf,
+        harden({ ...instanceRecord }),
+      );
+
+      addSeatObjPromiseKit.resolve(addSeatObj);
+      publicFacetPromiseKit.resolve(publicFacet);
+
+      // creatorInvitation can be undefined, but if it is defined,
+      // let's make sure it is an invitation.
+      if (creatorInvitation !== undefined) {
+        assert(
+          await invitationIssuer.isLive(creatorInvitation),
+          details`The contract did not correctly return a creatorInvitation`,
+        );
+      }
+
+      // Actually returned to the user.
+      return {
+        creatorFacet,
+        creatorInvitation: await creatorInvitation,
+        instance,
+        publicFacet,
+      };
```

Comment:
Another violation of the `await` restriction

```suggestion
      const creatorInvitation = await creatorInvitationP;
      // AWAIT
      return {
        creatorFacet,
        creatorInvitation,
        instance,
        publicFacet,
      };
```